src/language-markdown/printer-markdown.js

```diff
@@ -68,6 +68,9 @@ function genericPrint(path, options, print) {
 
   switch (node.type) {
     case "root":
+      if (node.tokens.length === 0 && node.children.length === 0) {
```

Comment:
There's no `tokens` attr in a standard [`root`](https://github.com/syntax-tree/mdast#root), `node.children.length === 0` should be enough.